src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Security.Tests
+{
+    internal static class SslStreamExtensions
+    {
+        public static Task AuthenticateAsClientAsync(this SslStream stream, SslClientAuthenticationOptions clientOptions,
+            bool async, CancellationToken cancellationToken = default)
+        {
+            return async
+                ? stream.AuthenticateAsClientAsync(clientOptions, cancellationToken)
+                : Task.Run(() => stream.AuthenticateAsClient(clientOptions), cancellationToken);
```

Comment:
Passing cancellation token to Task.Run is going to test Task.Run's cancellation ability; not AuthenticateAsClient.  Is that what's intended?  Should this be wrapping the call and registering with the token to dispose of the stream?  Or are there other tests doing that?

Reply:
No, this shouldn't be here. We decided not to support cancellation on the sync methods and solve it one level higher with the `stream.Dispose()`. I added few more targeted tests for Dispose during AuthenticateAs(Client/Server).